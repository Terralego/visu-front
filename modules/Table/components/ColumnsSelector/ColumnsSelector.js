import React from 'react';
import PropTypes from 'prop-types';
import { Button, Checkbox, Popover, Position, Intent } from '@blueprintjs/core';

export const ColumnsSelector = ({ columns, onChange, icon, position, intent }) => {
  const options = (
    <div className="table-columns-selector__options">
      {columns.map(({ value, display, label = value }, index) => (
        <Checkbox
          onChange={event => onChange({ event, index })}
          key={value}
          label={label}
          defaultChecked={display}
          value={value}
        />
      ))}
    </div>
  );
  return (
    <Popover
      content={options}
      position={position}
      popoverClassName="table-columns-selector__options-popover"
    >
      <Button
        icon={icon}
        minimal
        intent={intent}
      />
    </Popover>
  );
};

ColumnsSelector.propTypes = {
  columns: PropTypes.arrayOf(PropTypes.object).isRequired,
  onChange: PropTypes.func,
  icon: PropTypes.string,
  position: PropTypes.string,
  intent: PropTypes.string,
};

ColumnsSelector.defaultProps = {
  onChange () {},
  icon: 'properties',
  position: Position.RIGHT_BOTTOM,
  intent: Intent.PRIMARY,
};

export default ColumnsSelector;
