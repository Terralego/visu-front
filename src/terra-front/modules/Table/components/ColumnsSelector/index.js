import ColumnsSelector from './ColumnsSelector';

export default ColumnsSelector;
