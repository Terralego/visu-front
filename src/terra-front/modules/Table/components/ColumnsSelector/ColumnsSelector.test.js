import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import ColumnsSelector from './ColumnsSelector';

jest.mock('@blueprintjs/core', () => ({
  Button: jest.fn(() => null),
  Checkbox () { return null; },
  Label: jest.fn(() => null),
  Popover ({ children }) { return children; },
  Position: {
    RIGHT_BOTTOM: 'RIGHT_BOTTOM',
  },
  Intent: {
    PRIMARY: 'primary',
  },
}));

const props = {
  columns: [
    {
      value: 'Name',
      sortable: true,
      editable: true,
      display: true,
    },
    {
      value: 'Date',
      sortable: true,
      editable: true,
      format: { type: 'date', value: 'DD/MM/YYYY' },
      display: false,
    },
    {
      value: 'Lorem',
      sortable: true,
      editable: true,
      display: true,
    },
  ],
};

it('should render correctly', () => {
  const tree = renderer.create((
    <ColumnsSelector columns={props.columns} />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should change item', () => {
  const onChange = jest.fn();
  const wrapper = mount(<ColumnsSelector columns={props.columns} onChange={onChange} />, {});
  wrapper.find('Checkbox').get(1).props.onChange();
  expect(onChange).toHaveBeenCalled();
});

it('should toggle all', () => {
  const onChange = jest.fn();
  const wrapper = mount(<ColumnsSelector columns={props.columns} onChange={onChange} />, {});
  wrapper.find('Checkbox').get(0).props.onChange({ target: { checked: true } });
  expect(onChange).toHaveBeenCalledTimes(1);

  onChange.mockClear();
  wrapper.find('Checkbox').get(0).props.onChange({ target: { checked: false } });
  expect(onChange).toHaveBeenCalledTimes(2);
});
