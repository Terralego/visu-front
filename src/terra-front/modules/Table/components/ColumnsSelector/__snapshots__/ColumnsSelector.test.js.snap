// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="table-columns-selector__options"
/>
`;
